lib/Common/Memory/HeapBlock.inl

```diff
@@ -165,25 +165,34 @@ HeapBlock::UpdateAttributesOfMarkedObjects(MarkContext * markContext, void * obj
 
     if (attributes & TrackBit)
     {
-        FinalizableObject * trackedObject = (FinalizableObject *)objectAddress;
-
-#if ENABLE_PARTIAL_GC
-        if (!markContext->GetRecycler()->inPartialCollectMode)
+#ifdef RECYCLER_VISITED_HOST
+        if (GetHeapBlockType() == HeapBlock::HeapBlockType::LargeBlockType)
+        {
+            IRecyclerVisitedObject* recyclerVisited = static_cast<IRecyclerVisitedObject*>(objectAddress);
+            noOOMDuringMark = markContext->AddPreciselyTracedObject(recyclerVisited);
```

Comment:
Just realized that we're also going to fall into the !Leaf block below which adds the object to the normal marked objects stack. We should avoid that since it may end up producing false positives

Reply:
Solved this in LargeHeapBlock::UpdateAttributesOfMarkedObjects.

---
In reply to: [183829667](https://github.com/Microsoft/ChakraCore/pull/5034#discussion_r183829667) [](ancestors = 183829667)